src/expressions/dataTypes/casting/tryCastToType.ts

```diff
@@ -62,6 +62,10 @@ export function castToPrimitiveType(from: ValueType, to: ValueType): (value: Val
 		case ValueType.XSINTEGER:
 			return castToInteger(instanceOf);
 		case ValueType.XSNUMERIC:
+			// This case exists because numeric is a union type,
+			// therefore it does not fit into the inheritance hierarchy
+			// normally and needs to be turned into something that does
+			// (almost always a double) first.
```

Comment:
This comment doesn't make much sense, at least the last 2 lines

Reply:
Weird sentence, I'll rewrite it.